### PR TITLE
General entry: What is GraphQL?

### PR DESCRIPTION
**Pull request by EugeneGohh**
### Description

This pull request is related to issue #232 .

### Type of Change

- [x] Adding a new entry
- [ ] Editing an existing entry (fixing a typo, bug, issues, etc)
- [ ] Updating the documentation

### Checklist

- [x] My entry follows the Codecademy Docs style guide
- [x] I have performed a self-review of my own writing and code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked my entry and corrected any misspellings
- [x] All writings are my own

### CODE REVIEW

File: content/general/concepts/what-is-graphql/what-is-graphql.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+Title: 'What is GraphQL?' # This is the name that appears on the hub page for this language. Pay attention to capitalization and punctuation!
```
Comment (Dusch4593) [inline content/general/concepts/what-is-graphql/what-is-graphql.md:2]: We can remove the comment on this line.

File: content/general/concepts/what-is-graphql/what-is-graphql.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## Why adopt GraphQL?
+
+Before discussing GraphQL, developers should be familiar with REST. One of REST's disadvantages is that it can result in an excessive number of HTTP queries or too much data being fetched. GraphQL aims to improve upon REST philosophy by allowing developers to ask for a specific data instead of fetching a whole giant object. There is no room for errors like underfetching and overfetching with GraphQL. Finally, GraphQL allows developers to easily integrate with databases, JSON files, REST APIs and cloud services.
+
+## How GraphQL works under the hood?
+
+A typical yet simple GraphQL workflow can be seen below:
+
+- First, a specific data is requested from the client via a query to GraphQL server.
```
Comment (Dusch4593) [inline content/general/concepts/what-is-graphql/what-is-graphql.md:32]: Given we're using words like "First" and "Last", should we convert this list into an ordered, step-by-step list?

File: content/general/concepts/what-is-graphql/what-is-graphql.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  - 'Servers'
+CatalogContent:
+  - 'paths/computer-science'
+  - 'paths/back-end-engineer-career-path'
+  - 'paths/front-end-engineer-career-path'
+---
+
+## History of GraphQL
+
+GraphQL was developed by Facebook in 2012 to address internal issues with their mobile applications. As an alternative to typical REST architecture, GraphQL became open-source in 2015. Consequently, GraphQL makes data transfers more efficient, network usage for Facebook's mobile applications was drastically reduced. As of now, GraphQL has attracted a large number of developers and is becoming increasingly popular for client-server communication.
```
Comment (Dusch4593) [inline content/general/concepts/what-is-graphql/what-is-graphql.md:18]: > GraphQL was developed by Facebook in 2012...

Let's link out to the official GraphQL site, https://graphql.org/, on the first use of "GraphQL" in the entry.

File: content/general/concepts/what-is-graphql/what-is-graphql.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## What is GraphQL?
+
+GraphQL, an open-source query language as described in [GraphQL](graphql.org)'s official website. It allows developers to consume, construct, and create APIs in their own unique way.
```
Comment (Dusch4593) [inline content/general/concepts/what-is-graphql/what-is-graphql.md:22]: Let's remove the hyperlink here and replace it above (per my last comment). Also, 

> GraphQL, an open-source query language as described in GraphQL's official website.

I don't think this reads as a complete sentence. Could we, perhaps, incorporate the next sentence and rewrite it as "GraphQL is an open-source query language that allows developers to consume, construct, and create APIs in their own unique way".

File: content/general/concepts/what-is-graphql/what-is-graphql.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+## Why adopt GraphQL?
+
+Before discussing GraphQL, developers should be familiar with REST. One of REST's disadvantages is that it can result in an excessive number of HTTP queries or too much data being fetched. GraphQL aims to improve upon REST philosophy by allowing developers to ask for a specific data instead of fetching a whole giant object. There is no room for errors like underfetching and overfetching with GraphQL. Finally, GraphQL allows developers to easily integrate with databases, JSON files, REST APIs and cloud services.
+
+## How GraphQL works under the hood?
+
+A typical yet simple GraphQL workflow can be seen below:
```
Comment (Dusch4593) [inline content/general/concepts/what-is-graphql/what-is-graphql.md:30]: > A typical yet simple GraphQL workflow can be seen below:

I advise against using words like "simple" and "basic". We could, instead, use "common" to retain the spirit of "typical" and showcase beginner-friendliness as opposed to "simple".

File: content/general/concepts/what-is-graphql/what-is-graphql.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+## What is GraphQL?
+
+GraphQL, an open-source query language as described in [GraphQL](graphql.org)'s official website. It allows developers to consume, construct, and create APIs in their own unique way.
+
+## Why adopt GraphQL?
+
+Before discussing GraphQL, developers should be familiar with REST. One of REST's disadvantages is that it can result in an excessive number of HTTP queries or too much data being fetched. GraphQL aims to improve upon REST philosophy by allowing developers to ask for a specific data instead of fetching a whole giant object. There is no room for errors like underfetching and overfetching with GraphQL. Finally, GraphQL allows developers to easily integrate with databases, JSON files, REST APIs and cloud services.
```
Comment (Dusch4593) [inline content/general/concepts/what-is-graphql/what-is-graphql.md:26]: Solid pitch for GraphQL! 👏 👏

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
This will be a great addition to Docs! There are a few comments I left; some are grammar- and substance-related. Thank you for submitting this PR and let us know if you have any questions!

File: content/general/concepts/what-is-graphql/what-is-graphql.md (hunk 2 above)
Comment (EugeneGohh) [reply content/general/concepts/what-is-graphql/what-is-graphql.md:32]: Sure, I have converted to an ordered list.

**Review by Dusch4593 — APPROVED** [review]
LGTM!

**Comment by sonnynomnom** [comment]
Thanks, @EugeneGohh and @Dusch4593!

It's live: https://www.codecademy.com/resources/docs/general/what-is-graphql

I also made some changes: https://github.com/Codecademy/docs/commit/88f525d64f11a938ad9bc8f361efdd89de037954

**Pull request merged**